graylog2-server/src/main/java/org/graylog2/periodical/IndexRotationThread.java

```diff
@@ -88,7 +91,10 @@ public void initialize() {
     public void doRun() {
         // Point deflector to a new index if required.
         if (cluster.isConnected()) {
-            indexSetRegistry.forEach((indexSet) -> {
+            indexSetRegistry
+                    .getAll().stream()
+                    .filter(indexSet -> !isCurrentlyMigrated(indexSet))
+                    .forEach((indexSet) -> {
                 try {
                     if (indexSet.getConfig().isWritable()) {
                         checkAndRepair(indexSet);
```

Comment:
Aren't you afraid of race conditions?
I.e. in line 96 you may find out that index set is not migrated, go to line 100, spend some time in `checkAndRepair` and in a meantime... index set may start getting migrated...

Reply:
True, fixed, awesome review comment, thank you!